cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/DefaultDatasetNamespace.java

```diff
@@ -52,9 +53,14 @@ public String namespace(Id.Namespace namespaceId, String suffix) {
   }
 
   @Override
-  @Nullable
-  public Id.DatasetInstance fromNamespaced(String datasetInstanceName) {
-    return fromNamespaced(Id.DatasetInstance.from(Constants.SYSTEM_NAMESPACE, datasetInstanceName));
+  public Id.DatasetInstance fromNamespaced(String namespaced) {
+    Preconditions.checkArgument(namespaced != null, "Dataset name should not be null");
+    // Dataset name is of the format cdap.<namespace>.<dataset-name>
+    String invalidFormatError = String.format("Invalid format for dataset name '%s'. " +
+                                                "Expected - cdap.<namespace>.<dataset-name>", namespaced);
```

Comment:
I think the error message is wrong. We allow users to change the prefix by setting *dataset.table.prefix* to something else, so its not always guaranteed to start with "cdap.". So the error message should instead be 'Expected - <table-prefix>.<namespace>.<dataset-name>'